packages/sqlite/src/lib/Snapshot.ts

```diff
@@ -0,0 +1,27 @@
+import { StreamToken, ITimelineEvent } from "@equinox-js/core"
+import * as Token from "./Token.js"
+import { Format } from "./LibSqlClient.js"
+
+export const snapshotCategory = (original: string) => original + ":snapshot"
+
+export const streamName = (category: string, streamId: string) =>
+  `${snapshotCategory(category)}-${streamId}`
+export type Meta = { streamVersion: string }
+const streamVersion = (evt: ITimelineEvent<Format>) => {
+  const meta = JSON.parse(evt.meta ?? "null") as Meta | null
+  return meta ? BigInt(meta.streamVersion) : -1n
+}
+
+export const meta = (token: StreamToken): Meta => ({
+  streamVersion: String(Token.streamVersion(token)),
+})
+export async function decode<V>(
+  decode: (ev: ITimelineEvent<Format>) => Promise<V | null | undefined> | V | null | undefined,
+  events: ITimelineEvent<Format>[],
+): Promise<[StreamToken, V] | null> {
+  if (events.length > 0) {
+    const decoded = await decode(events[0])
```

Comment:
dont get the async - why?

Reply:
mechanical sympathy of a kind eh. (like second guessing the hybrid unit in your rally car)